server/src/main/java/org/elasticsearch/cluster/metadata/AutoExpandReplicas.java

```diff
@@ -148,17 +148,12 @@ public static Map<Integer, List<String>> getAutoExpandReplicaChanges(
                 if (autoExpandReplicas.enabled() == false) {
                     continue;
                 }
-                // Special case for stateless indices: auto-expand is disabled, unless number_of_replicas has been set
-                // manually to 0 via index settings, which needs to be converted to 1.
+                // Special case for stateless indices: ensures there is always at least one replica.
                 if (Objects.equals(
                     indexMetadata.getSettings().get(ExistingShardsAllocator.EXISTING_SHARDS_ALLOCATOR_SETTING.getKey()),
                     "stateless"
                 )) {
-                    if (indexMetadata.getNumberOfReplicas() == 0) {
-                        nrReplicasChanged.computeIfAbsent(1, ArrayList::new).add(indexMetadata.getIndex().getName());
-                    } else {
-                        continue;
-                    }
+                    autoExpandReplicas = new AutoExpandReplicas(1, autoExpandReplicas.maxReplicas(), true);
```

Comment:
I think we've tried similar things before and determined it did not work. I believe 0-all would try to expand to indexing nodes too without an allocation decider (maybe I missed something though, hope you can clarify then)?

Reply:
I believe this aspect is covered by 
https://github.com/elastic/elasticsearch/blob/ff45ad12ac6aee77aa2c466e401c21138a1bd3a5/server/src/main/java/org/elasticsearch/cluster/metadata/AutoExpandReplicas.java#L102-L112
as long as we have corresponding `shouldAutoExpandToNode` implemented.